src/main/java/htsjdk/variant/variantcontext/VariantContextBuilder.java

```diff
@@ -339,26 +367,53 @@ private void makeAttributesModifiable() {
      * filters can be <code>null</code> -&gt; meaning there are no filters
      *
      * @param filters Set of strings to set as the filters for this builder
+     *                This set will be copied so that external set can be
+     *                safely changed.
      * @return this builder
      */
     public VariantContextBuilder filters(final Set<String> filters) {
-        this.filters = filters;
+        if (filters == null) {
+            unfiltered();
+        } else {
+            this.filtersCanBeModified = true;
+            filtersAsIs(new HashSet<>(filters));
+        }
         return this;
     }
 
+    /**
+     * This builder's filters are set to this value
+     *
+     * filters can be <code>null</code> -&gt; meaning there are no filters
+     *
+     * @param filters Set of strings to set as the filters for this builder
+     * @return this builder
+     */
+    private void filtersAsIs(final Set<String> filters) {
```

Comment:
should this just be called set filters?

Reply:
well, it's private, and I wanted to be clear that it does less work than the .filters() method.